include/xtensor-zarr/xzarr_aws_store.hpp

```diff
@@ -156,27 +157,27 @@ namespace xt
         }
     }
 
-    xzarr_aws_stream xzarr_aws_store::operator[](const std::string& key)
+    xzarr_aws_stream xzarr_aws_store::operator[](const std::string& key) const
     {
         return xzarr_aws_stream((m_root + '/' + key).c_str(), m_bucket, m_client);
     }
 
-    void xzarr_aws_store::set(const std::string& key, const std::vector<char>& value)
+    void xzarr_aws_store::set(const std::string& key, const std::vector<char>& value) const
```

Comment:
I don't think the setter should be const (even if the implementation does not change the object, it is a bit strange to see a setter method const).

Reply:
Maybe it is strange, but it's true: the object is not changed, only some distant storage that is external to this class. Isn't it what `const` reflects?